csharp/src/Google.Protobuf.Test/FieldMaskTreeTest.cs

```diff
@@ -432,5 +432,44 @@ public void Merge(bool useDynamicMessage)
             Assert.IsNotNull(destination.Payload);
         }
 
+        [Test]
+        [TestCase(true, null)]
+        [TestCase(false, 24)]
+        public void MergeWrapperFields(bool replaceMessageFields, long? expectedInt64Value)
+        {
+            // Instantiate a destination with wrapper-based field types.
+            var destination = new TestWellKnownTypes()
+            {
+                StringField = "Hello",
+                Int32Field = 12,
+                Int64Field = 24,
+                BoolField = null,
+            };
+
+            // Set up a targeted update.
+            var source = new TestWellKnownTypes()
+            {
+                StringField = "Hi",
+                Int64Field = null
```

Comment:
I think we should possibly have two tests - one for "setting null values in wrapper fields" (for both StringField and Int64Field) and one for "setting non-null values in wrapped fields) (again, for both). We could potentially parameterize it all to have one test, but I suspect it's cleaner not to.

Reply:
Please see https://github.com/protocolbuffers/protobuf/pull/9602/commits/c4e7df18593b613ca08884fc5b955adae438ca3a

Splitting into different tests ended up producing too much repetition, specifically around the "make-sure-non-targeted-fields-remain-intact" part -- a part which I very much want to cover in all cases.

Went for a single test with a little more extensive test matrix instead.
In my opinion it seems clean enough without much unnecessary crud.

Reply:
Well, it certainly took me longer to review than a larger set of more focused tests would have :(